test/acceptance_with_python/rbac/test_schema.py

```diff
@@ -0,0 +1,118 @@
+        admin_client.collections.delete(name)
+
+    for permission in generate_missing_lists(permissions):
+        with role_wrapper(admin_client, request, permission):
+            with pytest.raises(weaviate.exceptions.UnexpectedStatusCodeException) as e:
+                custom_client.collections.create(name=name)
+            assert e.value.status_code == 403
+            assert "forbidden" in e.value.args[0]
+    admin_client.collections.delete(name)
+
+
+def test_rbac_collection_read(
+    admin_client, custom_client, role_wrapper: Role_Wrapper_Type, request: SubRequest
+):
+    name = _sanitize_role_name(request.node.name) + "col"
+    admin_client.collections.delete(name)
+    admin_client.collections.create(name=name)
+
+    with role_wrapper(admin_client, request, RBAC.permissions.config.read(collection=name)):
+        col = custom_client.collections.get(name=name)
+        assert col.config.get() is not None
+
+    with role_wrapper(admin_client, request, []):
+        col = custom_client.collections.get(name=name)
+        with pytest.raises(weaviate.exceptions.UnexpectedStatusCodeException) as e:
+            col.config.get()
+        assert e.value.status_code == 403
+        assert "forbidden" in e.value.args[0]
+    admin_client.collections.delete(name)
+
+
+def test_rbac_schema_read(
+    admin_client, custom_client, role_wrapper: Role_Wrapper_Type, request: SubRequest
+):
+    name = _sanitize_role_name(request.node.name) + "col"
+    admin_client.collections.delete(name)
+    admin_client.collections.create(name=name)
+
+    with role_wrapper(admin_client, request, RBAC.permissions.config.read()):
+        custom_client.collections.list_all()
+
+    with role_wrapper(admin_client, request, []):
+        with pytest.raises(weaviate.exceptions.UnexpectedStatusCodeException) as e:
+            custom_client.collections.list_all()
+        assert e.value.status_code == 403
+        assert "forbidden" in e.value.args[0]
+    admin_client.collections.delete(name)
+
+
+def test_rbac_collection_update(
+    admin_client, custom_client, role_wrapper: Role_Wrapper_Type, request: SubRequest
+):
+    name = _sanitize_role_name(request.node.name) + "col"
+    admin_client.collections.delete(name)
+    admin_client.collections.create(name=name)
+    permissions = [
+        RBAC.permissions.config.read(collection=name),
+        RBAC.permissions.config.update(collection=name),
+    ]
+    with role_wrapper(admin_client, request, permissions):
+        col_custom = custom_client.collections.get(name)
+        col_custom.config.update(description="test")
+
+    for permission in generate_missing_lists(permissions):
+        with role_wrapper(admin_client, request, permission):
+            col_custom = custom_client.collections.get(name)
+            with pytest.raises(weaviate.exceptions.UnexpectedStatusCodeException) as e:
+                col_custom.config.update(description="test")
```

Comment:
It took me some time to decypher the purpose of this block, it would be helpful to add a comment stating that in order to run an update we require read and update permissions. As that provides the context to understand why testing a role with only one of those permissions has to fail.

Reply:
I renamed the variables and the function, does that help? The setup of all tests is more or less the same "test with all permissions, then remove each one individually and make sure it fails"